Force smtplib to use IPv4 addresses

jsf9k opened the pull request:
@climber-girl noticed some cases where the text `[Errno 97] Address family not supported by protocol` appears in the `trustymail` output.  I dug into this, and it appears to happen because, when given a hostname to connect to, `smtplib` asks DNS for all A and AAAA records, tries them in succession, and uses the first one that works.

When running trustymail in AWS Lambda (at least in my VPC, which doesn't support IPv6) the `[Errno 97] Address family not supported by protocol` error appears when an IPv6 connection is attempted.  This makes the connection attempt fail hard, and the other IP addresses returned by DNS are not even attempted.

To get around this, I explicitly get an IPv4 address for the hostname and pass that to smtplib instead.  This way no IPv6 connections are ever attempted.

jsf9k committed:
Bump version from 0.7.2 to 0.7.3

jsf9k committed:
Force smtplib to only use IPv4 addresses

By default, smtplib looks for A and AAAA records from DNS and uses the
first one that it can connect to.  What I find when running in
Lambda (at least in my VPC that doesn't support IPv6) is that when DNS
returns IPv6 an address I get a low level "errno 97 - Address family
not supported by protocol" error and the other addresses returned by
DNS are not even attempted.  Therefore the hostname is not scanned at
all.

To get around this I look up the A record and use that instead of the
hostname in DNS when I call smtp_connection.connect().

jsf9k committed:
Remove print() statement that is not needed

dav3r commented on trustymail/trustymail.py:
```diff
@@ -157,10 +157,32 @@ def starttls_scan(domain, smtp_timeout, smtp_localhost, smtp_ports, smtp_cache):
+                    mail_server, port, socket.AF_INET, socket.SOCK_STREAM
+                )
+                sa = ans[0][4]
+                mail_server_ip_address = sa[0]
```
`ans` and `sa` variable names could be a bit more descriptive, but this is by no means a show-stopper.

jsf9k replied:
Yes, I was copying the names used in the python socket library source code.  I'll give them better names.

jsf9k replied:
Variable names improved in a81781b.

dav3r reviewed (approved):
Makes sense to me, though I left a nitpicky comment of my own.

jsf9k committed:
Give a few variables better names

Thanks to @dav3r for pointing this out.  I was using the names I saw
in the python socket library source code, but they needed improvement.
